### PR TITLE
Fix #1781: compact_items expects dict but sometimes passed list

### LINKED ISSUE #1781: mass.search service crashes when mass server response contains tracks from spotify
### What version of Music Assistant has the issue?

2024.1.0b2

### The problem

The problem is that `compact_item` in `services.py` tries to pop keys from a list when the dict it receives from the server contains a list of lists.

### How to reproduce

I don't really know, but with a spotify account setup, all searches fail when track is checked.

### Relevant log output

Here is a stack trace from hass:

```
homeassistant  | Traceback (most recent call last):
homeassistant  |   File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 26, in _handle_async_response
homeassistant  |     await func(hass, connection, msg)
homeassistant  |   File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 782, in handle_execute_script
homeassistant  |     script_result = await script_obj.async_run(
homeassistant  |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
homeassistant  |   File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 1587, in async_run
homeassistant  |     return await asyncio.shield(run.async_run())
homeassistant  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
homeassistant  |   File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 426, in async_run
homeassistant  |     await self._async_step(log_exceptions=False)
homeassistant  |   File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 479, in _async_step
homeassistant  |     self._handle_exception(
homeassistant  |   File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 502, in _handle_exception
homeassistant  |     raise exception
homeassistant  |   File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 476, in _async_step
homeassistant  |     await getattr(self, handler)()
homeassistant  |   File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 713, in _async_call_service_step
homeassistant  |     response_data = await self._async_run_long_action(
homeassistant  |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
homeassistant  |   File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 675, in _async_run_long_action
homeassistant  |     return long_task.result()
homeassistant  |            ^^^^^^^^^^^^^^^^^^
homeassistant  |   File "/usr/src/homeassistant/homeassistant/core.py", line 2149, in async_call
homeassistant  |     response_data = await coro
homeassistant  |                     ^^^^^^^^^^
homeassistant  |   File "/usr/src/homeassistant/homeassistant/core.py", line 2186, in _execute_service
homeassistant  |     return await target(service_call)
homeassistant  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
homeassistant  |   File "/config/custom_components/mass/services.py", line 69, in handle_search
homeassistant  |     compact_item(item)
homeassistant  |   File "/config/custom_components/mass/services.py", line 62, in compact_item
homeassistant  |     compact_item(subitem)
homeassistant  |   File "/config/custom_components/mass/services.py", line 56, in compact_item
homeassistant  |     item.pop(key, None)
homeassistant  | TypeError: pop expected at most 1 argument, got 2
```

And here is a problematic reply from the mass server, you can see that `external_ids` is a list of one element that itself contains a list:

```json
{
   "media_type":"track",
   "item_id":"1lCRw5FEZ1gPDNPzy1K4zW",
   "provider":"spotify",
   "name":"We Are The Champions",
   "provider_mappings":[
      {
         "item_id":"1lCRw5FEZ1gPDNPzy1K4zW",
         "provider_domain":"spotify",
         "provider_instance":"spotify",
         "available":true,
         "audio_format":{
            "content_type":"ogg",
            "sample_rate":44100,
            "bit_depth":16,
            "channels":2,
            "output_format_str":"ogg",
            "bit_rate":320
         },
         "url":"https://open.spotify.com/track/1lCRw5FEZ1gPDNPzy1K4zW",
         "details":"None"
      }
   ],
   "external_ids":[
      [
         "isrc",
         "GBUM71029619"
      ]
   ], ...
}
```

### Additional information

Seems like a server issue, it's strange that external_ids is a list!

### What version of Home Assistant Core are your running

2024.1.2

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Linux

### PR DESCRIPTION
**Pull request by jacobswe**
This pull request fixes #1781 by implementing a simple type check before invoking `compact_items`. I wasn't able to find any docs on preferences for contributing, so please shoot this back to me if any changes are needed!

**Summary:**
With Spotify connected, running the `mass.search` service causes an error to occur if `track` is passed as a `media_type`. This occurs because the `compact_items` function in `mass/services.py` expects and is type hinted to exclusively accept a dictionary, but when called from `handle_search` or recursively from within `compact_items`, no check is performed to ensure the correct type is passed. This usually does not cause errors, but in some circumstances lists are passed, in the case of the below example for the `external_ids` field.

**Recreation YAML:**
```yaml
service: mass.search
data:
  limit: 5
  name: Pink Floyd
  media_type:
    - track
```

**Error:**
The error `TypeError: pop expected at most 1 argument, got 2` is thrown because `list.pop` expects a single index whereas `dict.pop` expects a key and optionally a default value. This causes improper arguments to be passed when a list is given to `compact_items`.

**Working Local Demonstration:**
image

### CODE REVIEW

**Comment by jacobswe** [comment]
Hey HA music wizards, just wondering if I can do anything on this. Totally get you're probably busy but let me know! It would be cool to get this enabled on the public release.

File: custom_components/mass/services.py [hunk 1]
```diff
@@ -59,14 +59,16 @@ def compact_item(item: dict[str, Any]) -> dict[str, Any]:
                     item[key] = compact_item(value)
                 elif isinstance(value, list):
                     for subitem in value:
-                        compact_item(subitem)
+                        if isinstance(subitem, dict):
```
Comment (marcelveldt) [inline custom_components/mass/services.py:62]: It's always better to guard and breakout early. So in this case:

if not isinstance(subitem, dict):
   continue

Comment (marcelveldt) [reply custom_components/mass/services.py:62]: btw: maybe we also want to catch lists here ?

**Comment by jacobswe** [comment]
Thanks, happy to make that update, and I'll think about how to capture lists as well. I'll have time to push some changes on Sunday probably. Appreciate the review!

**Comment by jacobswe** [comment]
Pushed the guard clauses and verified working. I don't think anything is needed for lists here, they should just work I think? If required, can we do in separate merge?

**Comment by marcelveldt** [comment]
Thanks!

**Pull request merged**